numpy/lib/format.py

```diff
@@ -431,6 +433,14 @@ def _write_array_header(fp, d, version=None):
         header.append("'%s': %s, " % (key, repr(value)))
     header.append("}")
     header = "".join(header)
+    
+    # Add spare space for growth but make sure the array header size stays
+    # constant at all times.
```

Comment:
Can you expand on this note a bit, I think it might be cryptic without the background knowledge?
I.e. maybe mention something like: Add spare space to the header.  This allows appending data at the end of the file and rewriting the header taking into account new, larger, shape.

Reply:
As @rkern and me found out, just adding spare space does not really solve the problem. Example: adding 64 bytes of spare space every time might lead to larger headers than before if one tries to append to the array. So on append, one either needs to check how large the array head used to be and add spare space accordingly with some if-clause _or_ think about what the minimum and maximum size of the array header actually is. The minimum size is assumed with just a single character in the decimal representation of the shape, e.g. `(1, 5, 5)` for C order. The maximum size is some 20 digit stuff like `(12345678901234567890, 5, 5)`. Therefore, one needs to take the digit count of the growth axis into account, like when it's `(1, 5, 5)` add 20-1=19 bytes of spare space elsewhere (e.g. at the end of the header). For `(12345678901234567890, 5, 5)` it's 20-20=0 bytes. Therefore, the array header size stays constant at all times (no matter how small or large the array is in the growth dimension).

Reply:
Does anyone have a nice idea for a nicer comment?

Reply:
The point is, that the use-case is that we allow modifying the header in-place so we need "extra" space.  The reader can understand that I think, all we need is to say that someone may want to modify the file in-place to append data at the end.  In that case, the total header size must be constant.

Reply:
Very nice ideal, _in-place_ is actually what this is about! How about that: "Add some spare space so that the array header can be modified in-place when changing the array size, e.g. when growing it by appending data at the end"